### PR TITLE
Report more information in debugger for assertion failures etc.

### PR DESCRIPTION
**Pull request by jaykrell**
And cleanup a little.

32 times:

gdb:
```
 info registers
 info frame
 info locals
 up
```

lldb:
```
 reg red
 frame info
 frame variable
 up
```

### CODE REVIEW

File: mono/mini/mini-posix.c [hunk 1]
```diff
@@ -825,10 +825,14 @@ native_stack_with_gdb (pid_t crashed_pid, const char **argv, FILE *commands, cha
 	argv [3] = commands_filename;
 	argv [4] = "-nx";
 
-	fprintf (commands, "attach %ld\n", (long) crashed_pid);
-	fprintf (commands, "info threads\n");
-	fprintf (commands, "thread apply all bt\n");
-
+	fprintf (commands, "attach %ld\n"
```
Comment (jaykrell) [inline mono/mini/mini-posix.c]: Fewer calls to fprintf, smaller code.

File: mono/mini/mini-posix.c [hunk 2]
```diff
@@ -861,20 +870,16 @@ void
...
 		return;
 
 	commands = fopen (commands_filename, "w");
-	if (!commands) {
-		unlink (commands_filename);
-		return;
-	}
-
-	memset (argv, 0, sizeof (char*) * 10);
+	if (!commands)
```
Comment (jaykrell) [inline mono/mini/mini-posix.c]: Smaller code.

File: mono/mini/mini-posix.c [hunk 3]
```diff
@@ -861,20 +870,16 @@ void
 mono_gdb_render_native_backtraces (pid_t crashed_pid)
 {
 #ifdef HAVE_EXECV
-	const char *argv [10];
+	const char *argv [10] = { 0 };
```
Comment (jaykrell) [inline mono/mini/mini-posix.c]: Smaller source, probably same object code.

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/mini/mini-posix.c (hunk 1 above)
Comment (luhenry) [reply mono/mini/mini-posix.c]: This is changing code for changing code. Changing the number of calls to fprintf doesn't fix anything, it just looks better.

Comment (kumpera) [reply mono/mini/mini-posix.c]: Either way is fine.

Comment (jaykrell) [reply mono/mini/mini-posix.c]: It looks smaller, fewer function calls, less code, same functionality, so better.

Comment (vargaz) [reply mono/mini/mini-posix.c]: The improvement is really marginal here, so this is really just a style change.

**Comment by lewurm** [comment]
@monojenkins build Linux AArch64 Coop Suspend

**Comment by lewurm** [comment]
@monojenkins build Linux x64 Coop Suspend

**Comment by lewurm** [comment]
while I agree that there are some unnecessary code style changes in it, the actual feature implemented in this PR is worthwhile having in master, so let's get over with it and not be too hung up about it

File: mono/mini/mini-posix.c [hunk 4]
```diff
@@ -828,6 +828,12 @@ native_stack_with_gdb (pid_t crashed_pid, const char **argv, FILE *commands, cha
 	fprintf (commands, "attach %ld\n", (long) crashed_pid);
 	fprintf (commands, "info threads\n");
 	fprintf (commands, "thread apply all bt\n");
+	for (int i = 0; i < 32; ++i) {
```
Comment (jaykrell) [inline mono/mini/mini-posix.c]: This need only be one call to fprintf, not 4.

**Comment by lewurm** [comment]
@monojenkins build Linux ARMv7

**Comment by lewurm** [comment]
@monojenkins rebase

**Pull request merged**